src/todolist/TaskServiceImpl.java

```diff
@@ -1,43 +1,47 @@
 package todolist;
 
-import java.util.Locale;
+import java.util.Collection;
+import java.util.stream.Collectors;
 
 public class TaskServiceImpl implements TaskService {
-    private TaskRepository repository;
-
-    public TaskServiceImpl(TaskRepository repository) {
-        this.repository = repository;
-    }
-
-    @Override
-    public void addTask(String[] parameters) {
-        String title = String.join(" ", parameters);
-        String status = "open";
-        Task task = new Task(title, status);
-        repository.addTask(task);
-    }
-
-    @Override
-    public void listTask(Status status) {
-        String statusList = status.toString().toLowerCase(Locale.ROOT);
-        repository.listTask(statusList);
+  
+  private TaskRepository repository;
+  
+  public TaskServiceImpl(TaskRepository repository) {
+    this.repository = repository;
+  }
+  
+  @Override
+  public void addTask(String[] parameters) {
+    String title = String.join(" ", parameters);
+    Task task = new Task(title, Status.OPEN);
+    repository.addTask(task);
+  }
+  
+  @Override
+  public void closeTask(int id) {
+    Task task = repository.getTask(id);
+    if (task != null) {
+      task.setStatus(Status.CLOSED);
+      repository.closeTask(new Task(task.getTitle(), task.getStatus(), id));
```

Comment:
Посмотри, пожалуйста, на этот код внимательно. Это полезно проделывать code review самой себе. Тут очень странная логика, зачем ты тут выставляешь у полученной задачи статус, а потом создаешь копию задачи и в итоге даже не используешь задачу?
Думаю, что код тут можно было написать такой
```suggestion
      repository.save(task);
```